addons/crm/views/crm_lead_views.xml

```diff
@@ -332,6 +328,18 @@
                     <filter string="Exclude Opt Out"
                             name='not_opt_out' domain="[('opt_out', '=', False)]"
                             help="Leads that did not ask not to be included in mass mailing campaigns"/>
+                    <separator/>
+                    <filter string="My Activities" name="activities_my"
+                            domain="[('activity_ids.user_id', '=', uid)]"/>
+                    <filter string="Today Activities" name="activities_today"
+                            domain="[('activity_ids.date_deadline', '=', context_today().strftime('%Y-%m-%d'))]"/>
+                    <filter string="This Week Activities" name="activities_week"
+                            domain="['&amp;', ('activity_ids.date_deadline', '&gt;=', context_today().strftime('%Y-%m-%d')),
+                                     ('activity_ids.date_deadline', '&lt;=', (context_today()+datetime.timedelta(days=7)).strftime('%Y-%m-%d'))
+                            ]"/>
+                    <filter string="Overdue Activities" name="activities_overdue"
+                            domain="[('activity_ids.date_deadline', '&lt;', context_today().strftime('%Y-%m-%d'))]"
```

Comment:
Actually I will instead add a _search on activity_summary / type_id / user-id fields. With auto join it should be fast enough. And that way there aren't too many columns added.